src/queries/options.ts

```diff
@@ -12,3 +13,15 @@ export type TextMatchOptions = {
   exact?: boolean;
   normalizer?: NormalizerFn;
 };
+
+export type LegacyQueryOptions = {
+  legacy?: boolean;
+};
+
+export const shouldReturnCompositeComponent = (option: LegacyQueryOptions) => {
+  if (option.legacy === false) {
+    return false;
+  }
+
+  return option.legacy || getConfig().useLegacyQueries;
+};
```

Comment:
you can write it like that as well:
```suggestion
export const shouldReturnCompositeComponent = (option: LegacyQueryOptions) => option.legacy ?? getConfig().useLegacyQueries;
```

Reply:
It is indeed way more concise ! That part has been updated however since I removed legacy option